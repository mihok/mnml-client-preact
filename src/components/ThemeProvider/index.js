import { h, render, Component } from 'preact';
import './styles.css';

class ThemeProvider extends Component {
  getChildContext () {
    const { children, ...context } = this.props;
    return context;
  }
  render ({ children }) {
    return (children && children[0]) || null;
  }
}

export default ThemeProvider;
