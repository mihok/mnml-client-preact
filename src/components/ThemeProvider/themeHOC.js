import { h, render, Component } from 'preact';

const theme = (ComponentToWrap, context) => class ThemeComponent extends Component {
  render () {
    const { theme } = this.context;
    return <ComponentToWrap {...this.props} theme={theme} />;
  }
  };

export default theme;
