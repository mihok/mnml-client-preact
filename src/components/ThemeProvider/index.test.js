/* const render = require('preact-render-to-string');
 * const App = require('../thing');*/
import render from 'preact-render-to-string';
import { h } from 'preact';
import ThemeProvider from '.';

describe('App component', () => {
  it('should render ', () => {
    const tree = render(<ThemeProvider />);
    expect(tree).toMatchSnapshot();
  });
});
