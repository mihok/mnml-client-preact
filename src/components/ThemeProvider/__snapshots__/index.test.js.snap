// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render  1`] = `""`;
